src/frontend/src/common/queries/trees.ts

```diff
@@ -24,13 +24,13 @@ import { MutationCallbacks } from "./types";
 interface CreateTreePayload {
   name: string;
   samples: string[];
-  tree_type: string;
+  tree_type: string | undefined;
```

Comment:
yeah i was wondering if there was a way to have the `treeType` type be `undefined | string` and then have the `CreateTreePayload` `treeType` be a `string` since there will never be an occasion where we pass `undefined` in the payload.

Reply:
i couldn't find a clear way to do so 😞

Reply:
I think you'd have to make sure wherever the CreateTreePayload interface gets used -- or somewhere upstream of it where the payload gets generated and then also have any intermediate interfaces updated as well -- there's a type guard that's filtering out `undefined` cases.

very pseudocode
```
on handleSubmit
if (treeType === undefined) {
  // some kind of error handling or maybe just setShouldShowErrorNotification(true)
}
handle payload like usual here
```

Definitely feels like unnecessary boilerplate, but I'm pretty sure the type guard in that would let TS know it can trust it's not in the undefined case.

Alternatively, could have the interface let treeType be optional, but something in the mutation side raises an error that blows up the call before it actually POSTs to backend, but that's basically a matter of moving the boilerplate type guard.

Reply:
yeah that does seem like a lot of boilerplate for this.. i'll keep it as is for now